OpenRA.Game/Network/OrderIO.cs

```diff
@@ -31,6 +31,17 @@ public OrderPacket(MemoryStream data)
 			this.data = data;
 		}
 
+		public static OrderPacket Combine(IEnumerable<OrderPacket> packets, World world)
```

Comment:
Please change this to something like

```csharp
public static OrderPacket Combine(IEnumerable<OrderPacket> packets)
{
	var orders = new List<Order>();
	foreach (var packet in packets)
	{
		if (packet.orders == null)
			throw new InvalidOperationException("OrderPacket.Combine can only be used with locally generated OrderPackets.");

		orders.AddRange(packet.orders);
	}

	return new OrderPacket(orders.ToArray());
}
```

Order deserialization relies on the current world state, so `OrderPacket.GetOrders` will fail if we try to call it before the tick where we want to use them (this is a problem for save games, and in the future replays, where all parsing happens up front before any simulation).

We don't currently have any usecases where we would need to combine serialized order packets, so the simplest option is to make it explicit that we only support locally generated orders.